web-common/src/features/entity-management/file-artifacts.ts

```diff
@@ -97,6 +97,11 @@ export class FileArtifacts {
     return artifact;
   };
 
+  // Get existing artifact without creating a new one (for embed mode)
+  getExistingFileArtifact = (filePath: string): FileArtifact | undefined => {
+    return this.artifacts.get(filePath);
+  };
```

Comment:
In Cloud & embeds, we should be accessing the Canvas _resource_, never the _file_. We should read whatever spec we need from `resource.canvas.state.validSpec`. Is it possible to go that direction without a major refactor?